test/markup/typescript/comment-after-enclosed-lambda.expect.txt

```diff
@@ -0,0 +1,2 @@
+<span class="hljs-keyword">const</span> bad = <span class="hljs-function">(<span class="hljs-params">(a, b</span>) =&gt;</span> [...a, b]);
```

Comment:
This is not correct output, please compare to https://github.com/highlightjs/highlight.js/blob/master/test/markup/typescript/functions.expect.txt

The () aren't being wrapped or paired properly.  The proper fix here is likely more complex.

Reply:
You would also need some tests (if we don't have them) for nested parenthesis inside arguments, which is likely what the self was trying to catch in the first place.

```js
return this.sides.every((length,width=(3+2+(4/5))) => length > 0 );
```

Reply:
Yes, but Javascript has the same problem. At least this change produces the same output that JS language produces. What is the purpose of 'self' in Typescript params?

Reply:
In a perfect world something like I think:

```
const bad = ((a, b) => [...a, b]);
const bad = (a => [...a, b]);
```

```
bad = (<span class="hljs-function"><span class="hljs-params">(a, b)</span> =&gt;</span> [...a, b]);
bad = (<span class="hljs-function"><span class="hljs-params">a</span> =&gt;</span> [...a, b]);
```

Reply:
Ok, I see.

Reply:
> Yes, but Javascript has the same problem. At least this change produces the same output that JS language produces.

Then it would seem they are both broken, just in different ways.  Fixing some of these requires really digging in deep and understanding what is happening, why the rules are there, the different possible issues (like in this case nested parens), etc. My bet is this is one of the harder ones, but I"m not certain.

Reply:
I see you were trying to follow my last comment (thanks) but now that I've looked a little closer I don't think the fix here is just "do what JS does".

Reply:
```
------------v
const bad = ((a, b) => [...a, b]);
```

The problem here is that first ( is matched as the start of a function when it is not, it's just an expression.  That's what someone needs to tackle here - this misdetection.